src/main/java/it/gov/pagopa/payhub/auth/service/a2a/ClientServiceImpl.java

```diff
@@ -49,4 +50,11 @@ public Optional<Client> getClientByClientId(String clientId) {
 		return clientRetrieverService.getClientByClientId(clientId);
 	}
 
+	public ClientDTO verifyCredentials(String clientId, String clientSecret) {
+		return this.getClientByClientId(clientId)
+			.map(clientMapper::mapToDTO)
+			.filter(dto -> dto.getClientSecret().equals(clientSecret))
+			.orElseThrow(() -> new InvalidExchangeRequestException("Invalid client-credentials"));
```

Comment:
this is not an "ExchangeRequest", please, use another exception.
This is not an "invalid request" this is a forbidden operation